src/main/java/it/pagopa/pagopa/apiconfig/config/RequestFilter.java

```diff
@@ -0,0 +1,55 @@
+package it.pagopa.pagopa.apiconfig.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.UUID;
+
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@Slf4j
+public class RequestFilter implements Filter {
+
+    public static final String HEADER_REQUEST_ID = "X-Request-Id";
+
+    /**
+     * Get the request ID from the custom header "X-Request-Id" if present, otherwise it generates one.
+     * Set the X-Request-Id value in the {@code response} and in the MDC
+     *
+     * @param request http request
+     * @param response http response
+     * @param chain next filter
+     * @throws IOException if an I/O error occurs during this filter's
+     *                     processing of the request
+     * @throws ServletException if the processing fails for any other reason
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
```

Comment:
better to work before they reach the `DispatcherServlet` 👍 

![image](https://user-images.githubusercontent.com/36746022/146493270-68563e1d-f322-4361-a16f-94ea14b2eff2.png)